pkg/parquetquery/nilIter.go

```diff
@@ -0,0 +1,201 @@
+package parquetquery
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	pq "github.com/parquet-go/parquet-go"
+)
+
+// NilAttributeIterator copies all functions of the sync iterator with just the next() function being different
+type NilAttributeIterator struct {
+	syncIterator          SyncIterator
+	lastRowNumberReturned RowNumber
+	attrFound             bool
+	lastBuff              bool
+}
+
+var _ Iterator = (*NilAttributeIterator)(nil)
+
+func NewNilAttributeIterator(ctx context.Context, rgs []pq.RowGroup, column int, columnName string, readSize int, filter Predicate, selectAs string, maxDefinitionLevel int, opts ...SyncIteratorOpt) *NilAttributeIterator {
```

Comment:
The iterator has the parameters `columnName`, `readSize`, `filter`, `selectAs`, and `maxDefinitionLevel` but they are not use. In `makeNilIterFunc` you are also passing them down via `opts`. I think they should be removed